### PR TITLE
Initial talent API implementations

### PR DESCRIPTION
**Pull request by Numynum**

### CODE REVIEW

File: tools/sqlite.lua [hunk 1]
```diff
@@ -48,13 +48,13 @@ local function factory(theProduct)
   local function create(filename)
     local dbinit = { 'BEGIN' }
     for k, v in pairs(defs) do
-      table.insert(dbinit, ('CREATE TABLE %s (%s)'):format(k, table.concat(v.orderedfields, ',')))
+      table.insert(dbinit, ('CREATE TABLE %s ("%s")'):format(k, table.concat(v.orderedfields, '","')))
```
Comment (Numynum) [inline tools/sqlite.lua:51]: this is to handle column name `Index` 🙁

Comment (ferronn-dev) [reply tools/sqlite.lua:51]: oh this is a great fix. ideally it'd be in a separate PR but don't bother (yet)

File: wowless/env.lua [hunk 2]
```diff
@@ -6,7 +6,7 @@ local function dump(api)
   return function(...)
     for _, x in ipairs({ ... }) do
       d(x)
-      if api.UserData(x) then
+      if type(x) == 'table' and api.UserData(x) then
```
Comment (ferronn-dev) [inline wowless/env.lua:9]: thanks :)

File: data/state/Units.yaml [hunk 3]
```diff
@@ -6,10 +6,15 @@ value:
   guids:
     Player-1096-06DF65C1:
       class: 1
+      class_comment: Warrior
```
Comment (ferronn-dev) [inline data/state/Units.yaml:9]: nice idea!

**Review by ferronn-dev — APPROVED** [review]
Looks great!

**Pull request merged**